e2e_test/streaming/rate_limit/snapshot_amplification.slt

```diff
@@ -0,0 +1,43 @@
+# This test will test that barrier latency does not spike
+# when there's rate limit.
+# We test the case where snapshot read is amplified
+
+statement ok
+SET STREAMING_PARALLELISM=2;
+
+statement ok
+SET STREAMING_RATE_LIMIT=1;
+
+statement ok
+CREATE TABLE table (i1 int);
+
+statement ok
+INSERT INTO table select * from generate_series(1, 100000);
+
+statement ok
+flush;
+
+statement ok
+CREATE SINK sink AS
+    SELECT x.i1 as i1 FROM table x
+        JOIN table s1 ON x.i1 = s1.i1
+            JOIN table s2 ON x.i1 = s2.i1
+                JOIN table s3 ON x.i1 = s3.i1
+    WITH (connector = 'blackhole');
+
+# The following sequence of FLUSH should be fast, since barrier should be able to bypass sink.
+# Otherwise, these FLUSH will take a long time to complete, and trigger timeout.
+statement ok
+flush;
+
+statement ok
+flush;
+
+statement ok
+flush;
+
+statement ok
+drop sink sink;
+
+statement ok
+drop table table;
```

Comment:
Without this PR, this test will cause TLE, since snapshot read's chunk size is
minimally `chunk_size`, and gets significantly amplified.

Reply:
<img width="1487" alt="Screenshot 2023-12-07 at 10 33 29 AM" src="https://github.com/risingwavelabs/risingwave/assets/47273164/e10f7613-a636-47fa-8d8d-1ab47d29405e">